### PR TITLE
Emit an event if failed to sync labels to tikv stores

### PR DESCRIPTION
**Pull request by PengJi**
### What problem does this PR solve? 
fix: #2536

### Check List 

Tests 

 - Unit test

Code changes

 - Has Go code change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Emit an event if failed to sync labels to tikv stores
```

### CODE REVIEW

File: pkg/manager/member/tikv_member_manager.go [hunk 1]
```diff
@@ -725,7 +729,9 @@ func (tkmm *tikvMemberManager) setStoreLabelsForTiKV(tc *v1alpha1.TidbCluster) (
 		if !tkmm.storeLabelsEqualNodeLabels(store.Store.Labels, ls) {
 			set, err := pdCli.SetStoreLabels(store.Store.Id, ls)
 			if err != nil {
-				klog.Warningf("failed to set pod: [%s/%s]'s store labels: %v", ns, podName, ls)
+				evtStr := fmt.Sprintf("failed to set pod: [%s/%s]'s store labels: %v", ns, podName, ls)
+				tkmm.recorder.Event(tc, corev1.EventTypeWarning, FailedSetStoreLabels, evtStr)
+				klog.Warning(evtStr)
```
Comment (cofyc) [inline pkg/manager/member/tikv_member_manager.go:734]:
```suggestion
```
this is not necessary, the recorder will print logs too.

File: pkg/manager/member/tikv_member_manager.go [hunk 2]
```diff
@@ -725,7 +729,9 @@ func (tkmm *tikvMemberManager) setStoreLabelsForTiKV(tc *v1alpha1.TidbCluster) (
 		if !tkmm.storeLabelsEqualNodeLabels(store.Store.Labels, ls) {
 			set, err := pdCli.SetStoreLabels(store.Store.Id, ls)
 			if err != nil {
-				klog.Warningf("failed to set pod: [%s/%s]'s store labels: %v", ns, podName, ls)
+				evtStr := fmt.Sprintf("failed to set pod: [%s/%s]'s store labels: %v", ns, podName, ls)
```
Comment (cofyc) [inline pkg/manager/member/tikv_member_manager.go:732]:
```suggestion
				evtStr := fmt.Sprintf("failed to set labels %v for store (id: %d, pod: %s/%s): %v ", ls, store.Store.Id, ns, podName, err)
```

more informative

File: pkg/manager/member/tikv_member_manager.go [hunk 3]
```diff
@@ -725,7 +729,9 @@ func (tkmm *tikvMemberManager) setStoreLabelsForTiKV(tc *v1alpha1.TidbCluster) (
...
+				tkmm.recorder.Event(tc, corev1.EventTypeWarning, unHealthEventReason, evtStr)
```
[the remark this answers is not in the document]
Comment (PengJi) [reply pkg/manager/member/tikv_member_manager.go]: fixed

File: pkg/manager/member/tikv_member_manager.go (hunk 1 above)
Comment (PengJi) [reply pkg/manager/member/tikv_member_manager.go]: fixed

File: pkg/manager/member/tikv_member_manager.go (hunk 2 above)
Comment (PengJi) [reply pkg/manager/member/tikv_member_manager.go]: fixed

**Comment by cofyc** [comment]
/merge
Thanks! Great job!

**Comment by cofyc** [comment]
we had a problem in CI, will fix it soon

**Pull request merged**